src/main/java/org/opentestsystem/ap/ims/service/ItemManagerService.java

```diff
@@ -185,7 +196,74 @@ public ItemResponse beginEditTransaction(String itemId, String section, String c
     public void saveTransaction(String itemId, ItemChangeRequest request) {
         log.debug("save item {}", itemId);
         validator.validateSaveTransaction(itemId, request);
-        this.itemManager.saveItem(securityUtil.getItemBankUser(), request.getItem(), request.getMessage());
+        Item item = request.getItem();
+        updateServerManagedFields(item);
+        this.itemManager.saveItem(securityUtil.getItemBankUser(), item, request.getMessage());
+    }
+
+    private void updateServerManagedFields(Item item) {
+        if (item.getCore() == null || item.getCore().getMetadata() == null || item.getServerManaged() == null) {
+            return;
+        }
+
+        String itemId = item.getId();
+
+        ItemMetadata metadata = item.getCore().getMetadata();
+        String subject = metadata.getSubject();
+        String grade = metadata.getIntendedGrade();
+
+        ServerManaged serverManaged = item.getServerManaged();
+        serverManaged.setPrimaryEnhancedId(EMPTY);
+        serverManaged.setSecondaryEnhancedId(EMPTY);
+        serverManaged.setTertiaryEnhancedId(EMPTY);
+        serverManaged.setQuaternaryEnhancedId(EMPTY);
+
+        serverManaged.setPrimaryEnhancedId(contentSpecIdUtil.formatId(
+                itemId,
+                subject,
+                grade,
+                metadata.getPrimaryClaim(),
+                metadata.getPrimaryTarget(),
+                metadata.getPrimaryCommonCoreStandard(),
+                metadata.getPrimaryContentDomain(),
+                metadata.getPrimaryEmphasis()));
+
+        if (isMath(subject)) {
+            serverManaged.setSecondaryEnhancedId(contentSpecIdUtil.formatId(
+                    itemId,
+                    subject,
+                    grade,
+                    metadata.getSecondaryClaim(),
+                    metadata.getSecondaryTarget(),
+                    metadata.getSecondaryCommonCoreStandard(),
+                    metadata.getSecondaryContentDomain()));
+
+            serverManaged.setTertiaryEnhancedId(contentSpecIdUtil.formatId(
+                    itemId,
+                    subject,
+                    grade,
+                    metadata.getTertiaryClaim(),
+                    metadata.getTertiaryTarget(),
+                    metadata.getTertiaryCommonCoreStandard(),
+                    metadata.getTertiaryContentDomain()));
+
+            serverManaged.setQuaternaryEnhancedId(contentSpecIdUtil.formatId(
+                    itemId,
+                    subject,
+                    grade,
+                    metadata.getQuaternaryClaim(),
+                    metadata.getQuaternaryTarget(),
+                    metadata.getQuaternaryCommonCoreStandard(),
+                    metadata.getQuaternaryContentDomain()));
+        }
+    }
+
+    private boolean isMath(String subject) {
+        if (subject == null) {
+            return false;
+        }
+
+        return subject.trim().equalsIgnoreCase("math");
```

Comment:
Suggest looking to see if there is a constant already defined in `ItemConstants` that we could use rather than hardcoding the string here.